### PR TITLE
[SYCL][CUDA][WIP] PI API Sampler Properties Unit Test

### PR DESCRIPTION
**Pull request by StuartDAdams**
Provides testing for `pi_sampler` and `pi_sampler_properties`. 

[WIP] Requires #1993 to be merged first.

### CODE REVIEW

File: sycl/unittests/pi/cuda/test_sampler_properties.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
+//==---- PlatformTest.cpp --- PI unit tests --------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#define GTEST_HAS_COMBINE 1
```
Comment (alexbatashev) [inline sycl/unittests/pi/cuda/test_sampler_properties.cpp:9]: I guess we need this line due to a bit outdated Google Test in LLVM monorepo. But can we possibly use CMake's `target_compile_definitions` for that?

Comment (StuartDAdams) [reply sycl/unittests/pi/cuda/test_sampler_properties.cpp:9]: Sounds like a good idea to me. I'll take a look.

Comment (StuartDAdams) [reply sycl/unittests/pi/cuda/test_sampler_properties.cpp]: Done

File: sycl/unittests/pi/cuda/test_sampler_properties.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  pi_platform platform_;
+  pi_device device_;
+  pi_context context_;
+  pi_sampler sampler_;
+
+  pi_bool normalizedCoords_;
+  pi_sampler_filter_mode filterMode_;
+  pi_sampler_addressing_mode addressMode_;
+
+  SamplerPropertiesTest(){};
```
Comment (alexbatashev) [inline sycl/unittests/pi/cuda/test_sampler_properties.cpp:35]:
```suggestion
  SamplerPropertiesTest() = default;
```

**Comment by bader** [comment]
Should we remove [WIP] tag from this PR?

**Pull request merged**